typescript/lib/parsers/palm.ts

```diff
@@ -29,22 +69,75 @@ export class PaLMTextParser extends ParameterizedModelParser {
     options?: InferenceOptions | undefined,
     params?: JSONObject | undefined
   ): Promise<Output[]> {
-    throw new Error("Method not implemented.");
+    if (!this.client) {
+      const apiKey = getAPIKeyFromEnv("PALM_KEY");
+      this.client = new TextServiceClient({
+        authClient: new GoogleAuth().fromAPIKey(apiKey),
+      });
+    }
+
+    const completionParams = this.deserialize(prompt, aiConfig, params);
+    // Palm doesn't support streaming
+    const response = (await this.client.generateText(completionParams))[0];
```

Comment:
Why do we need to get 1st element? What does full response look like?

Reply:
Full response looks something like [response, request, {}]. I don't know what that last object is. 

Anyway, the destructuring here is needed to extract this response. Updated to use destructuring
`const [response] = await this.client.generateText(completionParams);`